### ISSUE TITLE
Issue with numpy>1.24

### ISSUE THREAD

**Issue by bknight1** [underworldcode/underworld2#652]
https://github.com/underworldcode/underworld2/blob/b72fc39337db6c36a68b585540262c14cc9fc59e/underworld/container/_indexset.py#L201

This produces:
```
File ~/Documents/Research/PyVenv/UW2/lib/python3.10/site-packages/underworld/container/_indexset.py:201, in IndexSet._addremove(self, indices, isadding)
    198 except:
    199     raise RuntimeError("An unknown error occurred relating to the object passed in.")
--> 201 self._AddOrRemoveWithNumpyArray( np.fromiter(indices, np.int), isadding )

File ~/Documents/Research/PyVenv/UW2/lib/python3.10/site-packages/numpy/__init__.py:284, in __getattr__(attr)
    281     from .testing import Tester
    282     return Tester
--> 284 raise AttributeError("module {!r} has no attribute "
    285                      "{!r}".format(__name__, attr))

AttributeError: module 'numpy' has no attribute 'int'
```

as numpy.int has been removed in 1.24.

**Comment by bknight1** [comment]
Alternatives are outlined in https://numpy.org/devdocs/release/1.20.0-notes.html

**Comment by julesghub** [comment]
Fixed with caf9ca0cb97f0f69fb9579166f35f9a24ad9709a
This will make the 2.14.2 release
